### PR TITLE
HPCC-14705 Thor rollup group loses records following skipped transform

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please review

File: thorlcr/activities/rollup/throllupslave.cpp [hunk 1]
```diff
@@ -591,25 +591,29 @@ class CRollupGroupSlaveActivity : public CSlaveActivity, public CThorDataLink
     CATCH_NEXTROW()
     {
         ActivityTimer t(totalCycles, timeActivities);
-        if (!eoi)
+        if (eoi)
+            return NULL;
+
+        loop
         {
             CThorExpandingRowArray rows(*this, queryRowInterfaces(input));
```
Comment (ghalliday) [inline thorlcr/activities/rollup/throllupslave.cpp]: I suspect this is inefficient recreating this row array each time - but I haven't checked.

**Comment by richardkchapman** [comment]
Bit scared to put a change like this into a point release. It's a bug fix, but who knows what code out there is going to change unexpectedly as a result...

File: thorlcr/activities/rollup/throllupslave.cpp (hunk 1 above)
Comment (jakesmith) [inline thorlcr/activities/rollup/throllupslave.cpp]: Likely to be when groups are small, though not new.
Could move out an call clearRows() instead, but will make that a separate PR.

**Comment by jakesmith** [comment]
@ghalliday - fix looks correct to me.

>Bit scared to put a change like this into a point release. It's a bug fix, but who knows what code out there is going to change unexpectedly as a result...

@richardkchapman - tend to agree, better not in a point release for that reason and there is a ECL coded workaround.

**Comment by richardkchapman** [comment]
I will leave for now pending 5.6 discussions

**Comment by richardkchapman** [comment]
@ghalliday  Please rebase to 5.6.0 and open new PR

**Comment by ghalliday** [comment]
Replaced with https://github.com/hpcc-systems/HPCC-Platform/pull/8096
